src/share/poudriere/common.sh

```diff
@@ -4671,6 +4682,20 @@ deps_fetch_vars() {
 	    shash_set pkgname-flavors "${_pkgname}" "${_flavors}"
 	[ -n "${_ignore}" ] && \
 	    shash_set pkgname-ignore "${_pkgname}" "${_ignore}"
+	if have_ports_feature SUBPACKAGES && [ -n "${subpackages}" ]; then
+		shash_set pkgname-subpackages "${_pkgname}" "${subpackages}"
+		for _spkg in ${subpackages}; do
+			local "pkgname_${_spkg}"
+			all_pkgnames=${all_pkgnames} PKGNAME.${_spkg} pkgname_${_spkg}
+		done
+		if ! port_var_fetch_originspec "${originspec}" \
+			${all_pkgnames} ; then
+			msg_error "Error looking up subpackage names for ${COLOR_PORT}${originspec}${COLOR_RESET}"
+			return 1
+		fi
+		# FIXME: all the package names for that port are in local
```

Comment:
Why not just fetch `PKGNAMES` in the `port_var_fetch_originspec` above where you added `SUBPACKAGES`? As far as I can tell the only use for `PKGNAMES` is to delete old packages for the incremental check. So they can all live in 1 variable. And having said that I don't see the use in Poudriere knowing `SUBPACKAGES`.

Reply:
Well, because I don't really know how poudriere works 😅
All I have been doing is trying stuff out until works 😄 
I don't know if poudriere would need to know the subpackages.  At one point though, people are going to be able to add a dependency on a subpackage (say php gets folded into a single port with many subpackages, and some other ports needs two extensions)